orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -104,15 +105,23 @@ class RunTaskHandler(
             registry
               .timeoutCounter(stage.execution.type, stage.execution.application, stage.type, taskModel.name)
               .increment()
-            task.onTimeout(stage)
-            throw e
+            taskResult = task.onTimeout(stage)
+
+            if ((taskResult?.status != FAILED_CONTINUE) &&
+              (taskResult?.status != TERMINAL)) {
```

Comment:
Don't need these wrapping parenthesis, do we?

Reply:
how do you like this?
 `if (!setOf(TERMINAL, FAILED_CONTINUE).contains(taskResult?.status))`

Reply:
the old me would be like... "i don't want to create a short-lived object on gen 0 gc heap" but that me is long gone

Reply:
if you're this kind of person you can make `setOf(TERMINAL, FAILED_CONTINUE)` a static field so you only ever have one